packages/office-ui-fabric-react/src/components/Stack/StackItem/StackItem.styles.ts

```diff
@@ -34,7 +34,10 @@ export const styles: IStackItemComponent['styles'] = (props, theme): IStackItemS
       align && {
         alignSelf: alignMap[align] || align
       },
-      className
+      className,
+      order && {
```

Comment:
let className be the last thing. The user's overrides should have highest specificity.

Reply:
(FWIW, we won't guarantee specificity for rules created outside of mergeStyles, but for classnames that mergeStyles created, it can destructure it and enforce specificity.